azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -272,74 +275,108 @@ public int getHttpProxyPort() {
+    }
+
+    protected Azure createAzureClient() throws AzureAuthFailureException, AzureExecutionException {
         try {
-            final AzureEnvironment environment;
-            if (!AzureAuthHelper.validateEnvironment(auth.getEnvironment())) {
-                Log.prompt(String.format(UNSUPPORTED_AZURE_ENVIRONMENT, auth.getEnvironment()));
-                environment = AzureEnvironment.AZURE;
-            } else {
-                environment = AzureAuthHelper.getAzureEnvironment(auth.getEnvironment());
+            MavenAuthConfiguration authConfiguration = auth;
+            if (authConfiguration == null) {
+                authConfiguration = new MavenAuthConfiguration();
+            }
+            authConfiguration.setType(getAuthType());
+            final String serverId = authConfiguration.getServerId();
+            if (StringUtils.isNotBlank(serverId)) {
+                authConfiguration = MavenSettingHelper.buildAuthConfigurationByServerId(session, settingsDecrypter, serverId);
             }
-            final String environmentName = AzureAuthHelper.getAzureEnvironmentDisplayName(environment);
-            if (environment != AzureEnvironment.AZURE) {
+            try {
+                validateAuthConfiguration(authConfiguration);
+            } catch (InvalidConfigurationException ex) {
+                final String messagePostfix = StringUtils.isNotBlank(serverId) ? ("in server: '" + serverId + "' at maven settings.xml.")
+                        : "in <auth> configuration.";
+                throw new AzureExecutionException(String.format("%s %s", ex.getMessage(), messagePostfix));
+            }
+            azureTokenWrapper = AzureAuthManager.getAzureTokenWrapper(authConfiguration).toBlocking().value();
+            if (Objects.isNull(azureTokenWrapper)) {
+                return null;
+            }
+            final AzureEnvironment env = azureTokenWrapper.getEnv();
+            final String environmentName = AuthHelper.getAzureEnvironmentDisplayName(env);
+            if (env != AzureEnvironment.AZURE) {
                 Log.prompt(String.format(USING_AZURE_ENVIRONMENT, environmentName));
             }
-            azureTokenWrapper = getAuthTypeEnum().getAzureToken(isAuthConfigurationExist() ? this.auth : null, environment);
-            final Azure azureClient = azureTokenWrapper == null ? null : AzureClientFactory.getAzureClient(azureTokenWrapper,
-                    this.subscriptionId, getUserAgent());
-
-            return azureClient;
-        } catch (IOException | AzureLoginFailureException e) {
+            if (StringUtils.isBlank(this.subscriptionId)) {
+                if (StringUtils.isNotBlank(azureTokenWrapper.getDefaultSubscriptionId())) {
+                    this.subscriptionId = azureTokenWrapper.getDefaultSubscriptionId();
+                } else {
+                    final TextIO textIO = TextIoFactory.getTextIO();
+                    final Azure tempAzure = Azure.configure()
```

Comment:
Could use `Azure.configure().authenticate($credential).subscriptions().list()` to get the subscriptions, and get the default subscription from credential